pkg/kubelet/types/pod_update.go

```diff
@@ -144,7 +146,17 @@ func (sp SyncPodType) String() string {
 // or equal to SystemCriticalPriority. Both the rescheduler(deprecated in 1.10) and the kubelet use this function
 // to make admission and scheduling decisions.
 func IsCriticalPod(pod *v1.Pod) bool {
-	return IsCritical(pod.Namespace, pod.Annotations) || (pod.Spec.Priority != nil && IsCriticalPodBasedOnPriority(*pod.Spec.Priority))
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodPriority) {
```

Comment:
We did not check priority feature gate before :) @dashpole , any background to share?

Reply:
Looks like that was an oversight when we added IsCriticalPodBasedOnPriority.  We should add it here.

Reply:
That's great ! 

LGTM then :)